app/builtin-pages/com/editor/file-tree.js

```diff
@@ -0,0 +1,279 @@
+const yo = require('yo-yo')
+import * as models from './models'
```

Comment:
Good work on this. +1 for building file-tree by copying from existing code; that was the right call.

I'm going to suggest we make a change to its structure.

Unless you plan for there to be multiple `FileTree` instances, let's change this from a class to just a regular module. We can just move any state contained by the class into a `// globals` section. Any methods can become exported API functions.

I suggest this because it's currently halfway between a module and a class, and I think it'll be clearer if we just go 100% module.